src/server/snap.rs

```diff
@@ -171,6 +174,28 @@ fn send_snap(
     res
 }
 
+/// Write binary snapshot to local disk, use RateLimiter to control the speed
+fn write_snapshot<W: Write>(w: &mut W, limiter: Arc<RateLimiter>, data: &[u8]) -> io::Result<()> {
+    let total = data.len();
+    let single = limiter.get_singleburst_bytes() as usize;
+    let mut curr = 0;
+    while curr < total {
+        let mut end;
+        if curr + single >= total {
+            end = total;
+            limiter.request((total - curr) as i64, 0);
+        } else {
+            end = curr + single;
+            limiter.request(single as i64, 0);
```

Comment:
I think the step should less than single burst limit, so others threads will not be blocked. BTW the step should has a maximum limitation, for example the RateLimiter is set as 100MB/s, and the refill period is 100ms, so single burst is 10MB, writing a 10MB batch to disk one time will affect other threads' write a lot, like writing WAL.

Reply:
Fixed.